izpack-installer/src/main/java/com/izforge/izpack/installer/bootstrap/InstallerGui.java

```diff
@@ -124,19 +124,32 @@ public void run()
 	            });
 	        }
 	        
-	        if (langCode == null)
-	        {
-	          installerContainer.getComponent(LanguageDialog.class).initLangPack();
-	        }
-	        else
-	        {
-	          installerContainer.getComponent(LanguageDialog.class).propagateLocale(langCode);
-	        }
-	        if (!installerContainer.getComponent(RequirementsChecker.class).check())
-	        {
-	            logger.info("Not all installer requirements are fulfilled.");
-	            installerContainer.getComponent(Housekeeper.class).shutDown(-1);
-	        }
+                final Exception[] guiThreadEx = new Exception[1];
+                SwingUtilities.invokeAndWait(new Runnable() {
+                    @Override
+                    public void run() {
+                        if (langCode == null)
+                        {
+                            try {
+                                installerContainer.getComponent(LanguageDialog.class).initLangPack();
+                            } catch (Exception ex) {
+                                guiThreadEx[0] = ex;
```

Comment:
> Just these two questions to understand some details:
>
>   * Why do you use an array here?
>   * Can't this be thrown directly, possibly having final actions in a finally clause?

In order to avoid that the exception is thrown on a different thread than before the change. Since there's the comment `// TODO not sure this works because thrown from AWT thread ...` at line ~70 I wonder how you handle exceptions in this multithreaded environment in general. Using an array to be able to assign values from another thread in a common workaround imo (I'll add a comment if that's fine). I can use a `FutureTask<Throwable>`, but maybe it's better to handle the exception on the EDT all along, like you said, but I'd like to make sure that this corresponds to your general multithreaded exception handling.

I have a bit the feeling that those exceptions are confused, based on the comment cited above which shouldn't be present in a stable release because it should be absolutely clear which effect the exception has as well as propagation of exceptions down from `main` (some are ignored, `catch`-blocks are too large sometimes). What do you think?